src/Stryker.Core/Stryker.Core/Mutants/MutationContext.cs

```diff
@@ -47,10 +56,28 @@ public MutationContext Clone()
             return new MutationContext(this);
         }
 
+        public void Discard()
+        {
+            if (HasStatementLevelMutant)
+            {
+                // some mutants 
+                Logger.LogDebug($"{BlockLevelControlledMutations.Count+StatementLevelControlledMutations.Count} mutations were not injected.");
+                foreach (var mutant in BlockLevelControlledMutations.Union(StatementLevelControlledMutations))
+                {
+                    mutant.ResultStatus = MutantStatus.CompileError;
+                    mutant.ResultStatusReason = "Stryker was not able to inject mutation in code.";
+                }
+                BlockLevelControlledMutations.Clear();
+                StatementLevelControlledMutations.Clear();
+            }
+
+        }
+
         public void Dispose()
         {
             if (_ancestor == null)
             {
+                Discard();
```

Comment:
Discard only decides to remove mutants that could not be injected if there are statement level mutants. That seems incorrect to me at Dispose time. Shouldn't ALL mutants that could not be injected be marked as error when there is no ancestor and the context is disposed?

Reply:
Well, all mutants are marked as error (i.e including blocklevel mutations). There are no 'expressionlevel' mutations backlog because those are injected on the fly.
Once again, this is a sign of sub par design, I agree with that view.